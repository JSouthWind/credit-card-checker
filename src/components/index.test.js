import React from 'react';
import Input from './';
import renderer from 'react-test-renderer';

let focusOnInputSpy = jest.fn();

jest
  .spyOn(Input.prototype, 'focusOnInput')
  .mockImplementation(focusOnInputSpy);


it('Input renders correctly', () => {
  const tree = renderer
    .create(<Input page="http://www.facebook.com" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(focusOnInputSpy).toHaveBeenCalled();
});