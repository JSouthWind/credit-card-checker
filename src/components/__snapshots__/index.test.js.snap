// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<div
  className="sc-bdVaJa ciFTBY"
>
  <div
    className="sc-bwzfXH gxYlwT"
  >
    <input
      className="sc-htpNat gdCsUx"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="0000 0000 0000 0000"
      type="text"
      value=""
    />
    <div
      className="sc-bxivhb dzCHsK"
      onClick={[Function]}
    >
      <span
        aria-hidden={true}
        className="fa fa-cc- fa-2x"
      />
    </div>
  </div>
</div>
`;
